### PR TITLE
DBZ-3989  trim string data before converts to JDBC value

### PR DESCRIPTION
**Pull request by blcksrx**
https://issues.redhat.com/browse/DBZ-3989

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Thx, @blcksrx! Re the fix, how about moving this to the default value converter class instead? Could you also:

* Add a test
* Verify what happens if there's a string with e.g. a letter, say '0a'? Will MySQL reject this one upon execution of the DDL?

**Comment by blcksrx** [comment]
I think its okay with this class. Im going to add test and verify that.

**Comment by blcksrx** [comment]
As I expected MySQL returns invalid default value error for these kinda situations:
```
MySQL [test]> CREATE TABLE test(id INTEGER DEFAULT '0a');
ERROR 1067 (42000): Invalid default value for 'id'
```

**Comment by gunnarmorling** [comment]
Just to make sure we're on the same page, I think the issue should be fixed in `MySqlDefaultValueConverter`, which deals with the default values obtained from MySQL.

> As I expected MySQL returns invalid default value error for these kinda situations

Ok, thanks for checking. So at least a basic level of sanity in MySQL here :)

**Comment by blcksrx** [comment]
@gunnarmorling

**Comment by blcksrx** [comment]
@gunnarmorling I wrote Test for it as you can see its the JDBC and Database engines behavior that converts and trim data. That's the reason I implemented on the core first. I can continue it also to other databases but I will wait for your response

File: debezium-core/src/test/java/io/debezium/jdbc/JdbcValueConvertersTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Test
+    @FixFor("DBZ-3989")
+    public void testConvertInteger() {
+        Column column = columnEditor.name("col1").type("INTEGER").jdbcType(Types.INTEGER).defaultValue("1 ").create();
+        Field field = new Field("test", -1, Schema.INT32_SCHEMA);
+        JdbcValueConverters converters = new JdbcValueConverters();
+
+        Object convertedData = converters.convertInteger(column, field, "1 ");
```
Comment (rk3rn3r) [inline debezium-core/src/test/java/io/debezium/jdbc/JdbcValueConvertersTest.java:35]: What happens with Strings like `abcABC `?

Comment (blcksrx) [reply debezium-core/src/test/java/io/debezium/jdbc/JdbcValueConvertersTest.java:35]: Actually this trim only applies on convert numerical stuff

Comment (rk3rn3r) [reply debezium-core/src/test/java/io/debezium/jdbc/JdbcValueConvertersTest.java:35]: Ah, then I was just irritated by `if (data instanceof String) {`. Never mind.

**Comment by blcksrx** [comment]
@gunnarmorling I added more tests and also did write tests for MySQL, shall I continue it for the others?

**Comment by jpechane** [comment]
@blcksrx I understand but still I believe this is something that belongs to default value converter, The value converters should get altready sanitized values. There are default value converters for other connectors present too.

**Comment by blcksrx** [comment]
@jpechane  You mean Instead of handling it on the `JDBCValueConvertor`, I should do it on the `DefaultValueConvertor` for each connector, right?

**Comment by jpechane** [comment]
@blcksrx Yes, and thinking about hat maybe a small refactoring giving all the defaultvalueconverters same parent would make also sense - so they can reuse a code.

**Comment by blcksrx** [comment]
Im going to close this PR and write it again.

**Pull request closed without merging**